### DISCUSSION TITLE
Rerun search with new parameters

### DISCUSSION THREAD

**Discussion by guhadebasish1** [TanStack/query#7434] [Q&A] [answered]
Hello! 

I have a page that displays various products. 
The page has a search box to search by product name as well as additional filters like sort by price, type etc.
Whenever the search criteria changes, the page should call the backend API endpoint passing in the latest search parameters to get the filtered list of products.

The various search parameters are maintained in Redux.

I am using useQuery to get list of products during initial load. 
I am wondering what approach should I take for rerunning the search when the search terms are updated? 

I tried invalidateCache but that reruns the useQuery with the original parameters.

Any advise please?

**Comment by TkDodo** [comment, marked as answer]
make the parameters part of the QueryKey
